chaindataloader/sideload.go

```diff
@@ -0,0 +1,109 @@
+package chaindataloader
+
+import (
+	"errors"
+	"fmt"
+	"io"
+
+	"github.com/btcsuite/btcd/wire"
+)
+
+var (
+	ErrUnsupportedSourceType = errors.New("source type is not supported")
+)
+
+// Reader is the interface a caller fetches data from.
+type Reader interface {
```

Comment:
This is called reader, but it doesn't have any methods that actually read/parse data.

Reply:
I think we can compress it into `BlockHeaderReader`?

Reply:
Maybe we can call it HeaderReader. Since it is for both blkhdr and cfhdr